docs/pages/api.rst

```diff
@@ -48,14 +48,38 @@ Signature: ``Collection::fromIterable(string $filepath): Collection;``
 
     Collection::fromFile('http://loripsum.net/api');
 
+fromGenerator
+~~~~~~~~~~~~~
+
+Create a collection from a `Generator`_.
+
+.. warning:: The difference between this constructor and ``fromIterable`` is that
+    the generator is decorated with a caching Iterator. ``Generators`` are not
+    **rewindable** by design and using ``fromIterable`` automatically adds the
+    caching layer for you.
+
+.. tip:: You can reproduce the same behavior by using ``fromIterable`` directly
+    followed by the ``cache`` operation.
+
+Signature: ``Collection::fromGenerator(Generator $generator): Collection;``
+
+.. code-block:: php
+
+    $generator = (static fn () => yield from range(1, 10))();
+    $generator->next();
+    $generator->next();
+
+    $collection = Collection::fromGenerator($generator);
+
 fromIterable
 ~~~~~~~~~~~~
 
 Create a collection from an iterable.
 
 .. warning:: When instantiating from a PHP `Generator`_, the collection object will inherit its behaviour:
     it will only be iterable a single time, and an exception will be thrown if multiple operations which attempt
-    to re-iterate are applied, for example ``count()``.
+    to re-iterate are applied, for example ``count()``. Use `Collection::fromGenerator()` to
+    circumvent this limitation.
```

Comment:
we could mention that another alternative is using `fromCallable` with a callable that returns the Generator

Reply:
Sure, it's done!